src/emulator/gui/src/profiles_manager_dialog.cpp

```diff
@@ -0,0 +1,189 @@
+    ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
+    ImGui::Begin("Profiles Manager", &gui.configuration_menu.profiles_manager_dialog, ImGuiWindowFlags_AlwaysAutoResize);
+    ImGui::PopStyleColor();
+    auto update_online_id = std::find(host.cfg.online_id.begin(), host.cfg.online_id.end(), gui.online_id);
+
+    // Online ID
+    ImGui::InputTextWithHint("##online_id", "Write your online ID", &gui.online_id);
+    if (ImGui::IsItemHovered())
+        ImGui::SetTooltip("Set your Online ID with maximum 16 character.");
+    ImGui::SameLine();
+    if (ImGui::Button("Add Online ID") && !gui.online_id.empty())
+        ImGui::OpenPopup("Online ID");
+    ImGui::PushStyleColor(ImGuiCol_Text, GUI_COLOR_TEXT_MENUBAR_OPTIONS);
+    if (ImGui::BeginPopupModal("Online ID", NULL, ImGuiWindowFlags_AlwaysAutoResize)) {
+        ImGui::PopStyleColor();
+        if (gui.online_id.length() > 16)
+            ImGui::Text("Error: Need reduce longer for online ID.\nActuall longer: %d.", gui.online_id.length());
+        else if (update_online_id != host.cfg.online_id.end())
+            ImGui::Text("Error: '%s' online ID already exist.", gui.online_id.c_str());
+        else
+            ImGui::Text("Succes: '%s' online ID added.", gui.online_id.c_str());
+
+        ImGui::Separator();
+        if (ImGui::Button("OK", ImVec2(128, 0))) {
+            if (update_online_id == host.cfg.online_id.end() && !(gui.online_id.length() > 16)) {
+                host.cfg.online_id.push_back(gui.online_id);
+                host.cfg.user_id += (int)host.cfg.online_id.size() - host.cfg.user_id;
+                gui.online_id.clear();
+            }
+            ImGui::CloseCurrentPopup();
+        }
+        ImGui::EndPopup();
+    } else
+        ImGui::PopStyleColor();
+
+    ImGui::PushItemWidth(300);
+    if (ImGui::ListBoxHeader("##online_id_list", (int)host.cfg.online_id.size(), 3)) {
+        for (auto i = 0; i < host.cfg.online_id.size(); i++) {
+            ImGui::Text("User ID: %02d, Online ID: %s", i, host.cfg.online_id[i].c_str());
+        }
+        ImGui::ListBoxFooter();
+    }
+    ImGui::PopItemWidth();
+    ImGui::Spacing();
+    if (host.cfg.online_id.size() > 1) {
+        ImGui::Combo("Select your User", &host.cfg.user_id,
+            [](void *vec, int idx, const char **list_online_id) {
+                auto &vector = *static_cast<std::vector<std::string> *>(vec);
+                if (idx < 0 || idx >= static_cast<int>(vector.size())) {
+                    return false;
+                }
+                *list_online_id = vector.at(idx).c_str();
+                return true;
+            },
+            static_cast<void *>(&host.cfg.online_id), (int)host.cfg.online_id.size());
+        ImGui::Spacing();
+        if (ImGui::Button("Delete Selected User"))
+            ImGui::OpenPopup("Delete User");
+        if (ImGui::IsItemHovered())
+            ImGui::SetTooltip("Obligatory keep one Online ID in same time.");
```

Comment:
What do you mean here?

Reply:
hmm nothingsd, is not needed now lol, i have just write for said obligatory need keep one user exist ^^